harness/determined/pytorch/_pytorch_context.py

```diff
@@ -168,16 +174,11 @@ def wrap_model(self, model: torch.nn.Module) -> torch.nn.Module:
             check.false(self._use_apex, "Must call wrap_model() before configure_apex_amp.")
 
             model = model.to(self.device)
-            if not self.distributed.size > 1 and self.n_gpus > 1:
-                check.eq(
-                    self._aggregation_frequency,
-                    1,
-                    "Please enable `optimized_parallel` to use aggregation "
-                    "frequency greater than 1 for single machine multi-GPU "
-                    "training.",
-                )
-                model = nn.DataParallel(model)
-                logging.debug("Initialized model for native parallel training.")
+
+            if self._distributed_backend.use_torch():
+                ddp_model = torch.nn.parallel.DistributedDataParallel(model)
+                self._wrapped_models[ddp_model] = model
+                model = ddp_model
```

Comment:
Code is usually more readable if you can avoid overwriting local variables.  This is a good example, where it would be harder to read the code wrong if you wrote it like this:

```python
if self._distributed_backend.use_torch():
    wrapped_model = torch.nn.parallel.DistributeDataParallel(model)
else:
    wrapped_model = model

# ... use wrapped_model hereafter
```